test/gtest/ucp/test_ucp_am.cc

```diff
@@ -511,6 +516,33 @@ UCS_TEST_P(test_ucp_am_nbx, zero_send)
 UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx)
 
 
+class test_ucp_am_nbx_eager_memtype : public test_ucp_am_nbx {
+public:
+    void init() {
+        modify_config("RNDV_THRESH", "inf");
+
+        if (!mem_buffer::is_gpu_supported()) {
+            UCS_TEST_SKIP_R("No GPU support");
+        }
```

Comment:
why need to have special test for GPU?
i'd just add memtype support to existing AM tests

Reply:
there are already many variants for am tests, also GPU is not supported for iovs. 
For rndv i want to test all combinations of send and recv memtypes with contig data only